thirdparty/spdlog/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+include(FetchContent)
+
+FetchContent_Declare(
+    spdlog
+    GIT_REPOSITORY https://github.com/gabime/spdlog.git
+    GIT_TAG        v1.5.0
+)
+
+FetchContent_MakeAvailable(spdlog)
```

Comment:
cc @jakirkham 

Would love your opinion about using FetchContent / ExternalProject in the context of conda builds and if this potentially presents any problems or challenges.

Reply:
Upgrading the version is definitely easier than the vendoring approach, arguably easier than git submodules too (see commit https://github.com/rapidsai/rmm/pull/369/commits/58b93cc02b7602633dcd5366c79e72076ac9d932).

Reply:
The primary difference would be in namespace resolution, and when the child project is acquired.
For `ExternalProject` the child is unaware of the parent, and source is acquired at build time.
For `FetchContent` the child is aware of the parent, and source is acquired at configure time.

The [documentation ](https://cmake.org/cmake/help/git-stage/module/FetchContent.html)covers some of the differences.

Reply:
Honestly with Conda my preference would be just to make `spdlog` a dependency. Would it be possible to check if `spdlog` is already available somewhere and skip this if so?

Reply:
@jakirkham I believe we're trying to address source builds that don't want to use conda.

I could be mistaken.

Reply:
My original intent was to pull in new releases of thrust/cub. I don't have a strong preference over conda vs cmake, but @harrism expressed concerns with relying on conda for c++ builds, thus this attempt at a conda alternative (for c++ dependency management).

Reply:
Yeah that all makes sense. Was more trying to say if the user already has `spdlog` (either with Conda or some other way), it would be nice to bypass the download. Otherwise agree this is a reasonable solution to ensure they are available.

Reply:
I think we should file this as an issue for future enhancement. Some of the build infrastructure improvements that @mt-jones has built a design doc for should address this.

Reply:
cmake FetchContent is designed for solving the problem where multiple libraries have the same dependency. I would like to see the mentioned design document.